src/librustc/ty/subst.rs

```diff
@@ -533,8 +533,7 @@ impl<'a, 'tcx> TypeFolder<'tcx> for SubstFolder<'a, 'tcx> {
                             data.name,
                             self.root_ty,
                             data.index);
-                        self.tcx.sess.delay_span_bug(span, &msg);
-                        r
+                        span_bug!(span, "{}", msg);
```

Comment:
What's the need for this change? The `delay_span_bug` approach hides bugs elsewhere from us, but also safely lets the compiler continue for the users' sake.

Reply:
Indeed, it hides bugs.